osdep/timer.c

```diff
@@ -73,11 +75,19 @@ int64_t mp_time_ns_add(int64_t time_ns, double timeout_sec)
     return time_ns + ti;
 }
 
-#if !HAVE_WIN32_INTERNAL_PTHREADS
 static int get_realtime(struct timespec *out_ts)
 {
 #if defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0
     return clock_gettime(CLOCK_REALTIME, out_ts);
+#elif defined(_WIN32)
+    union {
+        FILETIME ft;
+        ULARGE_INTEGER i;
+    } r;
```

Comment:
This nested union is funky, I guess it should work. I probably would do more explicit conversion tho, but I don't see any issue with current solution though.

Reply:
An union should be safe unless the struct members, their order or padding differs. Which according to docs is not the case.